### ISSUE TITLE
couchbase:change event not firing

### ISSUE THREAD

**Issue by jeffleus** [couchbaselabs/TodoLite-Ionic#1]
I used your project as an ionic starter template from this github repo.  When I run things, the 'couchbase:change' event does not seem to fire when a task item is added or deleted.  If I navigate away and back to the list, the new item appears (or, deleted item disappears).  So, the document add/delete is happening.  But, when I put a break on the $rootscope.$on('couchbase:change'...  It is never hit.  Thoughts?

**Comment by nraboy** [comment]
Sounds like the listener never started.  Can you confirm you have this line?:

https://github.com/couchbaselabs/TodoLite-Ionic/blob/master/www/js/app.js#L40

Also, what platform are we talking about? Android or iOS, and what version?

**Comment by jeffleus** [comment]
iOS 8.2 on an iPhone 6 Simulator

I connected Safari Web Inspector and placed a breakpoint at the referenced line (#40).  With the inspector attached, I ran the location.reload() in the console to try and hit the line.   But it throws a couple errors that may be because of the reload(), but it never hit the line.  Do you have a better way to but a breakpoint on that line to check w/ a simulator?  This is one of the challenges I find in debugging cordova apps...

[Error] Failed to load resource: the server responded with a status of 412 (Precondition Failed) (todo, line 0)
[Error] {"status":412,"error":"file_exists"}

**Comment by nraboy** [comment]
Yea, Apache Cordova apps are always a pain to debug, more so with iOS.  I think there may be a bug in the project, but here is a good test.

Can you add the `todoDatabase.listen()` inside the `basicLogin` function instead?  Replication seems to be fine from what you mentioned, so it just isn't broadcasting to AngularJS.  I think this will fix it.

**Comment by jeffleus** [comment]
Actually, I just bracketed the line w/log stmts (start listen and complete listen) and I see the messages post to the log w/out an error in between.  So, I think that line is successfully executing on start-up.  But, I will add the tododb.listen() in the login function and report back.  BTW, thx for the quick response.

**Comment by nraboy** [comment]
:+1:

**Comment by jeffleus** [comment]
Ugh.  Sorry to waste your time.  I forgot that I setup an AWS instance to support the SyncGateway feature during my testing.  And, the machine was offline.  I realized this once I saw the login code where the replicate lines are located.  When I brought it online and configured the SG correctly, things started working.  In fact, I emulated in both iOS and Android and could see "real-time" updates between the two clients.

But it does seem like poor design that the UI fails to properly update when the SG is down.  I will do some more testing when I get a chance tomorrow.  BTW, thanks for the tutorial.  Love the intro and really like the idea of Couchbase Mobile and SG to provide a great UX in mobile apps w/ data.
